WordPress/Classes/ViewRelated/NUX/SiteCreationCreateSiteViewController.swift

```diff
@@ -109,30 +151,119 @@ class SiteCreationCreateSiteViewController: NUXViewController {
-            handler: { [unowned self] _ in
-                self.navigationController?.dismiss(animated: true, completion: nil)
-        })
-
-        alertController.addAction(goHome)
-        present(alertController, animated: true, completion: nil)
+}
+
+// MARK: - Error Handling Extension
+
+private extension SiteCreationCreateSiteViewController {
+
+    // Possible views to direct the 'Try again' button to when validation errors occur.
+    enum DestinationViews {
+        case themeSelection
+        case details
+        case domainSuggestion
+        case createSite
     }
 
-    // MARK: - Navigation
+    /// Determines the error message displayed to the user depending on
+    /// the validation error that occurred.
+    ///
+    /// - Parameter validationError: validation error type returned by SiteCreationFields.validation
+    ///
+    func setErrorMessage(for validationError: SiteCreationFieldsError) {
+        switch validationError {
+        case .missingTitle:
+            setReturnViewController(for: .details)
+            errorMessage = NSLocalizedString("The Site Title is missing.", comment: "Error shown during site creation process when the site title is missing.")
+        case .missingDomain:
+            setReturnViewController(for: .domainSuggestion)
+            errorMessage = NSLocalizedString("The Site Domain is missing.", comment: "Error shown during site creation process when the site domain is missing.")
+        case .domainContainsWordPressDotCom:
+            setReturnViewController(for: .domainSuggestion)
+            errorMessage = NSLocalizedString("The Site Domain contains wordpress.com.", comment: "Error shown during site creation process when the site domain contains wordpress.com.")
+        case .missingTheme:
+            setReturnViewController(for: .themeSelection)
+            errorMessage = NSLocalizedString("The Site Theme is missing.", comment: "Error shown during site creation process when the site theme is missing.")
+        }
+    }
 
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        super.prepare(for: segue, sender: sender)
+    /// Determines the error message displayed to the user depending on the last status
+    /// reached in the site creation process, i.e the step it failed on.
+    ///
+    func setErrorMessageForLastStatus() {
+        guard let lastStatus = lastStatus else {
+            return
+        }
 
-        if let vc = segue.destination as? SiteCreationEpilogueViewController {
-            vc.siteToShow = newSite
+        errorMessage = {
+            switch lastStatus {
+            case .validating:
+                setReturnViewController(for: .domainSuggestion)
+                return NSLocalizedString("The Site Domain is invalid.", comment: "Error shown during site creation process when the site domain validation fails.")
+            case .gettingDefaultAccount:
+                setReturnViewController(for: .createSite)
```

Comment:
Heading back to the site creation VC isn't reattempting to create the site, but as we discussed it I think that's really easy to fix 😀